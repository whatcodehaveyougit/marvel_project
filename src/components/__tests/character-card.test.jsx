import { describe, expect, test } from "vitest"
import CharacterCard from "../character-card/character-card.componet"
import React from 'react'
import { render } from "@testing-library/react"
import { BrowserRouter as Router } from "react-router-dom";

const character = {
  id: '123123',
  name: 'The Hulk',
  thumbnail: {
    path: 'this-it-the-path',
    extension: 'jpg'
  },
  description: 'He is a big monstor'
}


describe('CharacterCardTests', () => {
  test('Does a snapshot of the CharacterCard component', () => {
    const { container } = render(
      <Router>
        <CharacterCard
          character={character}
        />
      </Router>

    )
    expect(container).toMatchSnapshot()

  })
})