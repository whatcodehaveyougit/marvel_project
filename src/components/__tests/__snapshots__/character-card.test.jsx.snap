// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CharacterCardTests > Does a snapshot of the CharacterCard component 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root character-card css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="The Hulk"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img thumbnail-image css-o69gx8-MuiCardMedia-root"
      src="this-it-the-path.jpg"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        The Hulk
      </h5>
      <div
        class="MuiGrid-root max-lines css-vj1n65-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          He is a big monstor
        </p>
      </div>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing card-actions css-1t6e9jv-MuiCardActions-root"
    >
      <a
        class="character-page-link"
        href="/character/123123"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Find out more
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
  </div>
</div>
`;
